### PR TITLE
chore: add custom properties example

### PR DESCRIPTION
**Pull request by smbea**
Context: form-js@0.7.0 blog post

### CODE REVIEW

File: custom-properties/customProperties.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// helpers 
+
+function hasExternalData(field) {
+  const {
+    properties
+  } = field;
+  return properties && properties.externalData
+}
+
+function mockFecth(_, _) {
```
Comment (nikku) [inline custom-properties/customProperties.js]:
```suggestion
function mockFetch(_, _) {
```

File: custom-properties/index.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    <!--
+      add your form schema to this script tag
+      alternatively, load it asynchronously from anywhere
+    -->
+    <script type="application/form-schema">
+    {
+      "schemaVersion": 1,
+      "exporter": {
+        "name": "form-js",
+        "version": "0.1.0"
```
Comment (nikku) [inline custom-properties/index.html]:
```suggestion
        "version": "0.7.0"
```

File: custom-properties/customProperties.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+})
+
+
+// helpers 
+
+function hasExternalData(field) {
+  const {
+    properties
+  } = field;
+  return properties && properties.externalData
```
Comment (nikku) [inline custom-properties/customProperties.js]:
```suggestion
  return properties && properties.externalData;
```

File: custom-properties/customProperties.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+      // (2.1) use custom property "externalData" to get external API endpoint
+      const endpoint = properties.externalData;
+      const res = mockFecth(endpoint, "GET");
+
+      // (2.2) map request response to field values
+      field.values = res.map(option => ({
+        value: option.id,
+        label: option.name
+      }))
```
Comment (nikku) [inline custom-properties/customProperties.js]:
```suggestion
      }));
```

File: custom-properties/index.html [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          }
+        },
+        {
+          "values": [],
+          "label": "Approved By",
+          "type": "select",
+          "id": "Field_3",
+          "key": "approvedBy",
+          "properties": {
+            "externalData": "lala"
```
Comment (nikku) [inline custom-properties/index.html]: Does `lala` make any sense? If not, make it a boolean?

**Review by nikku — CHANGES_REQUESTED** [review]
What is missing:

* README

Good start!

**Comment by nikku** [comment]
CC @ingorichtsmeier

File: custom-properties/index.html (hunk 5 above)
Comment (smbea) [reply custom-properties/index.html]: It doesn't. I just changed it to an example API endpoint

File: custom-properties/customProperties.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+// helpers 
+
+function hasExternalData(field) {
+  const {
+    properties
+  } = field;
+  return properties && properties.externalData;
+}
+
+function mockFetch(_, _) {
```
Comment (nikku) [inline custom-properties/customProperties.js]:
```suggestion
async function preloadData(url, method) {
```

File: custom-properties/customProperties.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+const container = document.querySelector('#form');
+
+const loadForm = FormViewer.createForm({
+  container,
+  schema
+});
+
+// (2) populate form fields with external data
+loadForm.then((form) => {
```
Comment (nikku) [inline custom-properties/customProperties.js]:
```suggestion
loadForm.then(async (form) => {
```

File: custom-properties/customProperties.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+  const {
+    components
+  } = schema;
+
+  components.forEach(field => {
```
Comment (nikku) [inline custom-properties/customProperties.js]: make a for of loop (so nested `await` works).

File: custom-properties/customProperties.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+      properties,
+      id,
+      type
+    } = field;
+
+    if (type === "select" && hasExternalData(field)) {
+
+      // (2.1) use custom property "externalData" to get external API endpoint
+      const endpoint = properties.externalData;
+      const res = mockFetch(endpoint, "GET");
```
Comment (nikku) [inline custom-properties/customProperties.js]:
```suggestion
      const res = await preloadData(endpoint, "GET");
```

**Review by nikku — COMMENTED** [review]
Suggestion to make the example more robust:

* add argument names to fetch
* make fetch async (this is what fetch is always going to be)

File: custom-properties/customProperties.js (hunk 8 above)
Comment (smbea) [reply custom-properties/customProperties.js]: `for` loop instead of `forEach`? May I ask why? -> Got it

**Comment by MaxTru** [comment]
Looks great!

* I tried it out and realized that nothing happens when clicking on `submit`. While this of course is not the focus of the example, I still think that this click on `submit` is a typical action a user would do. So consider to add a `console.log` with the submitted form data, when clicking submit.
* I would give the example index.html a meaningful title:
Screenshot_20220302_100208

**Comment by smbea** [comment]
@MaxTru I changed the html file name to viewer.html.
I added a `console.log` and also an `alert` just so the user has some visual feedback when submitting if they test the example.

File: custom-properties/viewer.html [hunk 10]
```diff
@@ -0,0 +1,89 @@
+<html>
+  <head>
+    <!--
```
Comment (MaxTru) [inline custom-properties/viewer.html:3]: consider to add
```html
  <title>Custom properties form-js example</title>
```

Comment (smbea) [reply custom-properties/viewer.html:3]: ✅

**Review by nikku — APPROVED** [review]
Looks good to me!

**Pull request merged**